app/utils/currencyConversion.ts

```diff
@@ -64,19 +64,33 @@ export const paymentAmountToText = (
   locale = "en-US",
 ): string => {
   if (paymentAmount.currency === WalletCurrency.USD) {
-    return (paymentAmount.amount / 100).toLocaleString(locale, {
-      style: "decimal",
-      maximumFractionDigits: 2,
-      minimumFractionDigits: 2,
-    })
+    try {
+      const amount = paymentAmount.amount / 100
+      if (amount) {
+        return amount.toLocaleString(locale, {
+          style: "decimal",
+          maximumFractionDigits: 2,
+          minimumFractionDigits: 2,
+        })
+      }
+      throw Error("unable to parse USD amount to text")
+    } catch (error) {
+      console.error(error)
+      return "USD amount parsing error"
+    }
   }
 
   if (paymentAmount.currency === WalletCurrency.BTC) {
-    return paymentAmount.amount.toLocaleString(locale, {
-      style: "decimal",
-      maximumFractionDigits: 0,
-      minimumFractionDigits: 0,
-    })
+    try {
+      return paymentAmount.amount.toLocaleString(locale, {
+        style: "decimal",
+        maximumFractionDigits: 0,
+        minimumFractionDigits: 0,
+      })
+    } catch (e) {
+      console.error(e)
+      return "Unable to parse amount to"
```

Comment:
Here you can return "BTC formatting error"

Reply:
> Here you can return "BTC formatting error"
I think the issue here is  how this interpolates with the return statement in the parent function `paymentAmountToTextWithUnit` in [line 52](https://github.com/GaloyMoney/galoy-mobile/blob/046d285fc46dd4c3009ffa933984a93845607ff7/app/utils/currencyConversion.ts#L52)